examples/templates/gcp-devcontainer/main.tf

```diff
@@ -144,45 +277,50 @@ resource "google_compute_instance" "vm" {
     # The startup script runs as root with no $HOME environment set up, so instead of directly
     # running the agent init script, create a user (with a homedir, default shell and sudo
     # permissions) and execute the init script as that user.
-    startup-script = <<-META
-    #!/usr/bin/env sh
-    set -eux
+    startup-script = local.startup_script
+  }
+}
 
-    # If user does not exist, create it and set up passwordless sudo
-    if ! id -u "${local.linux_user}" >/dev/null 2>&1; then
-      useradd -m -s /bin/bash "${local.linux_user}"
-      echo "${local.linux_user} ALL=(ALL) NOPASSWD:ALL" > /etc/sudoers.d/coder-user
-    fi
+# Create a Coder agent to manage the workspace.
+resource "coder_agent" "dev" {
+  count              = data.coder_workspace.me.start_count
+  arch               = "amd64"
+  auth               = "token"
+  os                 = "linux"
+  dir                = "/workspaces/${trimsuffix(basename(data.coder_parameter.repo_url.value), ".git")}"
+  connection_timeout = 0
```

Comment:
Does setting this to zero actually serve a purpose? Doesn't it just use the default of 30s in this case?

Reply:
Unsure; this was from the original template IIRC.